src/core/Form/Toggle/ToggleButton/ToggleButton.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { Component } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { InternalTokensProp, TokensProp } from '../../../theme';
+import { AutoId } from '../../../../utils/AutoId';
+import { getConditionalAriaProp } from '../../../../utils/aria';
+import { Text } from '../../../Text/Text';
+import { HtmlSpan, HtmlButton, HtmlButtonProps } from '../../../../reset';
+import { withSuomifiDefaultProps } from '../../../theme/utils';
+import { ToggleBaseProps, baseClassName } from '../ToggleBase/ToggleBase';
+import { baseStyles } from './ToggeButton.baseStyles';
+import { ToggleIcon } from '../ToggleBase/ToggleIcon';
+
+const toggleClassNames = {
+  disabled: `${baseClassName}--disabled`,
+  button: `${baseClassName}--button`,
+  label: `${baseClassName}_label`,
+  iconContainer: `${baseClassName}_icon-container`,
+};
+
+export interface ToggleButtonProps
+  extends ToggleBaseProps,
+    Omit<HtmlButtonProps, 'onClick'> {
+  /** Event handler to execute when clicked */
+  onClick?: ({ toggleState }: { toggleState: boolean }) => void;
+}
```

Comment:
Should we also have **soft disable** for ToggleButton as we have for Button?

Reply:
I don't see it as necessary for this as with the normal button. For normal button it was added to mainly support e.g. submit buttons when there is mandatory functionality that can be disabled due to errors. If toggle is disabled it should not be an issue for screen readers to not show it at all. Or maybe not, might add some value as well.

Reply:
Will leave this as future improvement in case needed.